File: practice/algorithms/implementation/new_year_chaos/new_year_chaos.py
#!/usr/bin/env python

def count_less_then(i_q, i_n):
    cnt = 0
    for POS in xrange(i_n - 1, -1, -1):
        if i_q[POS] - POS - 1 > 2:
            return -1
        for OPOS in xrange(max(0, i_q[POS] - 2), POS):
            if i_q[POS] < i_q[OPOS]:
                cnt += 1
    return cnt

t = int(raw_input().strip())
for _ in xrange(t):
    n = int(raw_input().strip())
    q = map(int, raw_input().strip().split())
    cnt = count_less_then(q, n)
    if cnt >= 0:
        print cnt
    else:
        print "Too chaotic"

File: practice/algorithms/implementation/new_year_chaos/new_year_chaos.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int count_less_then(const vector<int> & i_q)
{
    int res = 0;
    for (int pos = i_q.size() - 1; pos >= 0; --pos)
    {
        if (i_q[pos] - 1 - pos > 2)
        {
            return -1;
        }
        for (int opos = max(0, i_q[pos] - 2); opos < pos; ++opos)
        {
            if (i_q[pos] < i_q[opos])
            {
                res++;
            }
        }
    }
    
    return res;
}

int main()
{
    int t, n, res;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        vector<int> q(n);
        for (int pos = 0; pos < n; ++pos)
        {
           cin >> q[pos];
        }
        
        res = count_less_then(q);
        if (res >= 0)
        {
            cout << res << endl;
        }
        else
        {
            cout << "Too chaotic" << endl;    
        }
        
    }
    
    return 0;
}

